src/modules/azure-azcopy-sync.test.ts

```diff
@@ -0,0 +1,477 @@
+            // Arrange
+            delete localFileOptions.destination.path;
+            calculatedCommandArgs[2] = _getBlobStorageUrl(
+                localFileOptions.destination
+            );
+
+            // Act
+            AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalledWith(
+                command,
+                calculatedCommandArgs,
+                spawnSyncArgs
+            );
+        });
+
+        test("when destination.sasToken provided then used in the destination URL argument", () => {
+            // Arrange
+            localFileOptions.destination.sasToken = "other_test_sas_token";
+            calculatedCommandArgs[2] = _getBlobStorageUrl(
+                localFileOptions.destination
+            );
+
+            // Act
+            AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalledWith(
+                command,
+                calculatedCommandArgs,
+                spawnSyncArgs
+            );
+        });
+
+        test("when destination.sasToken not provided then not used in the destination URL argument", () => {
+            // Arrange
+            delete localFileOptions.destination.sasToken;
+            calculatedCommandArgs[2] = _getBlobStorageUrl(
+                localFileOptions.destination
+            );
+
+            // Act
+            AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalledWith(
+                command,
+                calculatedCommandArgs,
+                spawnSyncArgs
+            );
+        });
+
+        test(`when fails then it calls shell.exit`, async () => {
+            // Arrange
+            const exitCode = faker.random.number({ min: 1 });
+            const spawnSyncSpy = TestUtils.spyOnSpawnSync(exitCode);
+
+            // Act
+            await AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalled();
+            expect(shellExitSpy).toHaveBeenCalledWith(exitCode);
+        });
+    });
+
+    // #endregion localFile
+
+    // -----------------------------------------------------------------------------------------
+    // #region containers
+    // -----------------------------------------------------------------------------------------
+
+    describe("containers", () => {
+        let containerOptions: SyncContainerOptions;
+        let calculatedCommandArgs: string[];
+
+        beforeEach(() => {
+            containerOptions = {
+                deleteDestination: false,
+                destination: {
+                    account: "destination_test_account",
+                    container: "destination_test_container",
+                    path: "destination_test_path",
+                    sasToken: "?destination_test_sas_token=test",
```

Comment:
I know this is for testing, but is `sasToken` going to be a query param when this command is being used? If so, it might be useful to rename this variable to better indicate that it is a query param containing the sasToken

Reply:
After reviewing this again, the SAS is a series of query parameters.  Azure provides us with this as the entire query string.  See the following as an example.  `?sv=[sv]&ss=[s]&srt=[srt]&sp=[sp]&se=[se]&st=[st]&spr=[spr]&sig=[sig]`.  I'm going to leave it like this for simplicity so we don't have to pass these properties in individually.